docs/release-notes/next/PT-161380219-check-for-reentrant-calls.md

```diff
@@ -0,0 +1 @@
+* Disallows reentrant calls (i.e. contract calls into a currently executing contract) in Sophia.
```

Comment:
I see documented [not-yet-implemented attach transaction](https://github.com/aeternity/protocol/blame/a8fc667df4b777ebbb7d5edbffe377d6ce3a085a/contracts/contract_transactions.md#L6). Does this PR answer any of the following questions?
* Is contract call transaction with caller equal to recipient allowed?
* Does contract create transaction allow initialization code to call back to sender of contract create transaction?

Reply:
At the moment we allow calls to the origin address (i.e. the signer of the top-level call transaction), see `aect_call_tx:new/1` which sets the call stack to empty if not present. Same for `aect_create_tx`. This PR doesn't take an active position on whether we should allow this, however.